platform/view/core/manager/context.go

```diff
@@ -269,6 +269,10 @@ func (ctx *ctx) ResetSessions() error {
 	return nil
 }
 
+func (ctx *ctx) PutService(service interface{}) error {
+	return ctx.localSP.RegisterService(service)
+}
+
```

Comment:
While I strongly believe that there should exist a `PutX` if we already have a `GetX` function, I believe call it `RegisterService` makes more sense to me :D